evm/combine-signatures.js

```diff
@@ -0,0 +1,273 @@
+                    return;
+                }
+
+                totalWeight += validatorWeight;
+
+                batchSignatures.push(signature);
+
+                if (totalWeight >= threshold) {
+                    break;
+                }
+            }
+
+            if (totalWeight < threshold) {
+                printError('Total signer weight less than threshold', totalWeight);
+                return;
+            }
+
+            const proof = defaultAbiCoder.encode(
+                ['address[]', 'uint256[]', 'uint256', 'bytes[]'],
+                [validatorAddresses, weights, threshold, batchSignatures],
+            );
+
+            const authAddress = address || contracts.AxelarGateway?.authModule;
+            const auth = new Contract(authAddress, IAxelarAuth.abi, wallet);
+
+            let isValidProof;
+
+            try {
+                isValidProof = await auth.validateProof(expectedMessageHash, proof);
+            } catch (error) {
+                printError('Invalid batch proof', error);
+                return;
+            }
+
+            if (!isValidProof) {
+                printError('Invalid batch proof');
+                return;
+            }
+
+            const input = defaultAbiCoder.encode(['bytes', 'bytes'], [message, proof]);
+
+            printInfo('Batch input data for gateway execute function', input);
+
+            break;
+        }
+
+        case 'executeBatch': {
+            const { input } = options;
+
+            validateParameters({ isValidCalldata: { input } });
+
+            const contractName = 'AxelarGateway';
+
+            const gatewayAddress = address || contracts.AxelarGateway?.address;
+            const gateway = new Contract(gatewayAddress, IAxelarGateway.abi, wallet);
+
+            const gasOptions = await getGasOptions(chain, options, contractName);
+
+            const tx = await gateway.execute(input, gasOptions);
+
+            await handleTx(tx, chain, gateway, action, 'Executed');
+
+            break;
+        }
+
+        default: {
+            throw new Error(`Unknown signature action ${action}`);
+        }
+    }
+}
+
+async function main(options) {
+    await mainProcessor(options, processCommand);
+}
+
+if (require.main === module) {
+    const program = new Command();
+
+    program.name('combine-signatures').description('script to combine manually created signatures and construct gateway batch');
+
+    addBaseOptions(program, { address: true });
+
+    program.addOption(
+        new Option('--action <action>', 'signature action').choices(['computeMessageHash', 'constructBatch', 'executeBatch']),
+    );
+    program.addOption(new Option('-m, --message <message>', 'bytes message (validators sign the hash of this message)').env('MESSAGE'));
+    program.addOption(new Option('-i, --input <input>', 'batch input consisting of bytes message (data) and bytes proof').env('INPUT'));
```

Comment:
```suggestion
    program.addOption(new Option('-i, --batchData <input>', 'batch input consisting of bytes message (data) and bytes proof').env('INPUT'));
```
need a readable name